app/serializers/user_bookmark_serializer.rb

```diff
@@ -35,7 +35,15 @@ def topic
   end
 
   def post
-    @post ||= object.post || Post.unscoped.find(object.post_id)
+    if for_topic?
+      @post ||= topic.first_post
```

Comment:
hmmm so this will change to the "last read" post later?

Reply:
Yes those changes will be in the next PR, this is mainly so the user bookmark list doesn't blow up. However that does bring up the question: When showing and searching topic-level bookmarks in the user bookmark list, should we be using the first post content, or the last unread post content? If the latter searching in https://github.com/discourse/discourse/pull/14272/files/45019bea922e24db615180c419e92ce8af0889a8#diff-0f7c6a210ef52b10b632c103844a35108a837501df63a47d2f7e077cf95b878f is a bit more complex